src/types/mpi.rs

```diff
@@ -28,23 +29,36 @@ const MAX_EXTERN_MPI_BITS: u32 = 16384;
 /// );
 /// ```
 ///
-pub fn mpi(input: &[u8]) -> nom::IResult<&[u8], MpiRef<'_>> {
+pub fn mpi(input: &[u8]) -> IResult<&[u8], MpiRef<'_>> {
     let (number, len) = be_u16(input)?;
 
     let bits = u32::from(len);
     let len_actual = (bits + 7) >> 3;
 
     if len_actual > MAX_EXTERN_MPI_BITS {
-        Err(Err::Error(error_position!(
-            input,
-            nom::ErrorKind::Custom(errors::MPI_TOO_LONG)
-        )))
+        /*
+            Err(Err::Error(
+                error_position!(
+                input,
+                nom::error::ErrorKind::Custom(errors::MPI_TOO_LONG)
+            )))
+            FIXME should this give more information in the error?
+            FIXME is this the right Error Variant?
+        */
+        Err(Err::Error(Error::InvalidInput))
     } else {
         // same as take!
         let cnt = len_actual as usize;
         match number.slice_index(cnt) {
-            None => nom::need_more(number, nom::Needed::Size(cnt)),
-            Some(index) => {
+            Err(needed) => {
+                // if input.at_eof() {
+                //     Err(nom::Err::Failure(nom::error::Error::new(input, nom::error::ErrorKind::Eof)))
+                // } else {
+                // FIXME what about eof?
```

Comment:
what about this?

Reply:
Right, I wanted to check what todo about that.
Turns out for slices at_eof was always false in nom 4.
So this is correct as is.
I have removed the FIXME in 44da300

Reply:
great